src/main/java/com/firebolt/jdbc/type/BaseType.java

```diff
@@ -159,8 +174,22 @@ private <T> T transform(StringToColumnTypeConversion conversion) throws SQLExcep
 		if (isNull(conversion.getValue())) {
 			return null;
 		}
-		//noinspection unchecked
-		return (T) transformFunction.apply(conversion);
+		for (int i = 0; i < transformFunctions.length - 1; i++) {
+			try {
+				//noinspection unchecked
+				return (T) transformFunctions[i].apply(conversion);
+			} catch (RuntimeException e) {
+				if (!shouldTryFallback.test(conversion.getValue())) {
+					throw new FireboltException(e.getMessage(), e, TYPE_TRANSFORMATION_ERROR);
+				}
+			}
+		}
+		try {
+			//noinspection unchecked
+			return (T) transformFunctions[transformFunctions.length - 1].apply(conversion);
+		} catch (RuntimeException e) {
+			throw new FireboltException(e.getMessage(), e, TYPE_TRANSFORMATION_ERROR);
+		}
```

Comment:
Can't we merge the for loop and the last transform case? It should be just the matter of modifying the if condition. I think this way it will be easier to understand.

Reply:
Done. You are right - this version is simpler